src/transformers/quantizers/quantizer_hqq.py

```diff
@@ -91,6 +91,65 @@ def validate_environment(self, *args, **kwargs):
             else:
                 self.using_multi_gpu = len(set(device_map.values())) > 1
 
+    def update_missing_keys(
+        self, model: "PreTrainedModel", missing_keys: List[str], prefix: str, **kwargs
+    ) -> List[str]:
+        if self.pre_quantized:
+            return [key for key in missing_keys if ("weight" not in key)]
+        else:
+            return missing_keys
+
+    # Adds missing keys for HQQLinear modules that are loaded but the model with initialized with torch.nn.Linear
+    def update_expected_keys(
```

Comment:
I am not sure I understand exactly why we need to do this post loading vs while loading + is gonna be quite annoying to maintain as it feels like there's a lot of hacks but fine for me!

Reply:
Unfortunately, this was the only way to make this work. The issue is that, when the model is created, it creates `nn.Linear` layers that have a "weight" parameter, but there's no "weight" parameter in `HQQLinear`, so when it tries to load the `HQQLinear` state-dict, it breaks.  I agree it's a bit complicated, actually the whole `HQQLinear` support has been complicated.

I think one better way of doing this in the future is to have a native `QuantLinear` layer officially supported by HF transformers, which could potentially integrate any quant method. I pitched this idea to Marc as well, I think it's worth thinking about. Happy to assist with that as well.